opentech/apply/funds/views.py

```diff
@@ -68,22 +76,54 @@ def get_queryset(self):
         return self.filterset_class._meta.model.objects.current().for_table(self.request.user)
 
     def get_context_data(self, **kwargs):
-        kwargs = super().get_context_data(**kwargs)
-
         search_term = self.request.GET.get('query')
-        kwargs.update(
+
+        return super().get_context_data(
             search_term=search_term,
             filter_action=self.filter_action,
+            **kwargs,
         )
 
-        return super().get_context_data(**kwargs)
+
+@method_decorator(staff_required, name='dispatch')
+class BatchUpdateReviewersView(DelegatedViewMixin):
+    form_class = BatchUpdateReviewersForm
+    context_name = 'batch_reviewer_form'
+
+    def save_all(self, request, *args, **kwargs):
+        """
+        Loop through all submissions selected on the page,
+        Add any reviewers that were selected,  only if they are not
+        currently saved to that submission.
+        Send out a message of updates.
+        """
+        for submission_id in request.POST["submission_ids"].split(","):
+            submission = ApplicationSubmission.objects.get(id=submission_id)
+
+            old_reviewers = set(submission.reviewers.all())
+            reviewers = User.objects.filter(id__in=request.POST.getlist('staff_reviewers'))
```

Comment:
We probably want to be doing this outside the loop. so we only make the request for users once.